specification/hugr.md

```diff
@@ -1433,27 +1433,34 @@ conversion to/from the binary serialised form.
 We propose the following simple serialized structure, expressed here in
 pseudocode, though we advocate MessagePack format in practice (see
 [Serialization implementation](serialization.md)).
-Note in particular that node and port weights are stored as separate
-maps to the graph structure itself, and that hierarchical relationships
-have a special encoding outside `edges`, as an optional parent field
-(the first) in a node definition. `Operation` refers to serialized
-payloads corresponding to arbitrary `Operations`. Metadata could also be
-included as a similar map.
+Note in particular that hierarchical relationships
+have a special encoding outside `edges`, as a field `parent`
+in a node definition. 
+
+The other required field in a node is `op` which identifies an operation by
+name, and is used as a discriminating tag in validating the remaining fields.
+The other fields are defining data for the particular operation.
+Metadata could also be included as a map keyed by node index.
 
 ```rust
 struct HUGR {
-  nodes: [Node]
-  edges: [Edge]
-  node_weights: map<Int, Operation>
+  nodes: [Node],
+  edges: [Edge],
 }
 
-// (parent, #incoming, #outgoing)
-struct Node = (Optional<Int>, Int, Int)
+struct Node{
+  // parent node
+  parent: Int,
+  // name of operation
+  op: String
+  //other op-specific fields
+  ...
```

Comment:
args (for custom ops)?

Reply:
are you suggesting that field should be mandated on all nodes or that it should
be mentioned as a possible extra field?

Reply:
Yeah currently it's part of the `...`
Maybe we should come back to this once we know what the spec for defining args in this format is? #186 and any follow on. I could leave an issue?

Reply:
Yeah OK; agreed we can't necessarily/completely define that now. OTOH perhaps mentioning `args` as a field would say what we do know now even without completing the picture with the bits we don't yet. Ok, up to you

Reply:
Yeah I've added a line above.